Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -182,210 +173,468 @@ internal void OnClientChangeScene(string scenePath, SceneOperation sceneOperatio
 
-            if (logger.LogEnabled()) logger.Log("ClientScene.Ready() called.");
+            if (logger.LogEnabled()) logger.Log("[NetworkSceneManager] - ClientScene.Ready() called.");
 
             // Set these before sending the ReadyMessage, otherwise host client
             // will fail in InternalAddPlayer with null readyConnection.
-            Client.Player.IsReady = true;
+            Client.Player.SceneIsReady = true;
 
             // Tell server we're ready to have a player object spawned
             Client.Player.Send(new ReadyMessage());
         }
 
         #endregion
 
-        #region Server
+        #region Server Side
 
-        // called after successful authentication
-        void OnServerAuthenticated(INetworkPlayer player)
+        /// <summary>
+        ///     Allows server to fully load new scene or additive load in another scene.
+        /// </summary>
+        /// <param name="scenePath">The full path to the scenes files or the names of the scenes.</param>
+        /// <param name="players">List of player's we want to send the new scene loading or unloading to.</param>
+        /// <param name="shouldClientLoadOrUnloadNormally">Should client load or unload the scene in normal non additive way</param>
+        /// <param name="sceneOperation">Choose type of scene loading we are doing <see cref="SceneOperation"/>.</param>
+        private void ServerSceneLoading(string scenePath, INetworkPlayer[] players,
+            bool shouldClientLoadOrUnloadNormally, SceneOperation sceneOperation = SceneOperation.Normal)
         {
-            logger.Log("NetworkSceneManager.OnServerAuthenticated");
+            if (string.IsNullOrEmpty(scenePath))
+            {
+                throw new ArgumentNullException(nameof(scenePath),
+                    "[NetworkSceneManager] - ServerChangeScene: " + nameof(scenePath) + " cannot be empty or null");
+            }
 
-            player.Send(new SceneMessage { scenePath = ActiveScenePath, additiveScenes = additiveSceneList.ToArray() });
-            player.Send(new SceneReadyMessage());
+            //if(players == null || players.Length == 0)
+            //    throw new ArgumentNullException(nameof(scenePath),
+            //        "[NetworkSceneManager] - list of player's cannot be null or no players.");
+
+
+            if (logger.LogEnabled()) logger.Log("[NetworkSceneManager] - ServerChangeScene " + scenePath);
+
+            // Let server prepare for scene change
+            logger.Log("[NetworkSceneManager] - OnServerChangeScene");
+
+            ServerStartedSceneChange?.Invoke(scenePath, sceneOperation);
+
+            if (!Server.LocalClientActive)
+                LoadSceneAsync(scenePath, players, sceneOperation).Forget();
+
+            // notify all clients about the new scene
+            if (players.Length > 0 && shouldClientLoadOrUnloadNormally)
```

Comment:
looks like only thing that change in the if/else is the `SceneOperation` so would be better to do the SendToMany call once after the if/else

Reply:
This also is dependant on the shouldClientLoadOrUnloadNormally option that end users can supply. So I dont think we can do that.